contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol

```diff
@@ -44,26 +44,29 @@ contract OCR2DRRegistry is
     // There are only 1e9*1e18 = 1e27 juels in existence, so the balance can fit in uint96 (2^96 ~ 7e28)
     uint96 balance; // Common LINK balance that is controlled by the Registry to be used for all consumer requests.
     uint96 blockedBalance; // LINK balance that is reserved to pay for pending consumer requests.
+    uint32 pendingRequestCount; // pendingRequestCount used to prevent a subscription with pending requests from being deleted
   }
   // We use the config for the mgmt APIs
   struct SubscriptionConfig {
     address owner; // Owner can fund/withdraw/cancel the sub.
     address requestedOwner; // For safely transferring sub ownership.
     // Maintains the list of keys in s_consumers.
     // We do this for 2 reasons:
-    // 1. To be able to clean up all keys from s_consumers when canceling a subscription.
+    // 1. To be able to clean up s_isAuthorizedConsumer when canceling a subscription.
     // 2. To be able to return the list of all consumers in getSubscription.
-    // Note that we need the s_consumers map to be able to directly check if a
-    // consumer is valid without reading all the consumers from storage.
     address[] consumers;
   }
-  // Note a nonce of 0 indicates an the consumer is not assigned to that subscription.
-  mapping(address => mapping(uint64 => uint64)) /* consumer */ /* subscriptionId */ /* nonce */
-    private s_consumers;
+
+  // Use a contract-wide nonce instead of a different nonce per consumer for simplicity
```

Comment:
Nonce per customer avoids pretty much any possibility of ever overflowing the nonce. Though it is pretty unreasonable to do so as a shared nonce because of how much they can hold.